### ISSUE TITLE
see if the jp translations can be given automatic furigana using MeCab

### ISSUE THREAD

**Issue by Pomax** [Pomax/BezierInfo-2#60]
more of a "neat to have" than a necessity, of course. Also of course the question of whether or not MeCab can be made to play nice with Node.js at all.

**Comment by Pomax** [comment]
@mashabow plenty of novels, textbooks, and dictionaries on my bookshelf have 振り仮名, but I don't really have any science publications - is 振り仮名 even a thing in those? Would it make sense to have the primer text come with it?

**Comment by mashabow** [comment]
Yes, science books and papers also have some 振り仮名. There are not so many difficult/uncommon words in the primer, so I'd like to add 振り仮名 only for some unfamiliar technical terms (e.g. convex hull; 凸包 とつほう) and it is enough for Japanese readers, I think.

**Comment by Pomax** [comment]
That seems reasonable. We can probably figure out the nicest way to represent that in the markdown when we run across them. There is the HTML ruby notation, but for markdown that seems a bit over the top. I've used western parentheses next to the kanjiform in the past, so that might work here, too.

**Comment by mashabow** [comment]
How about this?

- syntax: http://conv.denshochan.com/markdown#ruby
- implementation: https://github.com/denshoch/DenDenMarkdown

**Comment by Pomax** [comment]
Not a fan of the curly braces (they mean too many things all over the place) but I'm down for `（漢字、かんじ）の（書、か）き（方、かた）`

**Comment by mashabow** [comment]
I understand, and now almost agree with you. A couple of Japanese parentheses and commas are already used in our text; I am just concerned about it.

FYI, there is yer another notation with `《》` and `｜`, which is relatively popular in online novels.

- https://kakuyomu.jp/help/entry/notation
- http://www.aozora.gr.jp/annotation/etc.html#ruby

**Comment by Pomax** [comment]
I was going to do pattern matching on "opening parens" + "kanji codepoint" + "japanese comma" + "kana codepoints" + optional("japanese comma" + "kana codepoints") + "closing parens" which I don't believe is used anywhere in normal text.
